### PR TITLE
Avoid abbreviations in API

### PR DESCRIPTION
**Pull request by MabezDev**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [ ] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [ ] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

### Pull Request Details 📖

I went through our stable drivers and retroactively applied this rule. Spi changes are handled in https://github.com/esp-rs/esp-hal/pull/2833.

### CODE REVIEW

File: esp-hal/src/i2c/master/mod.rs [hunk 1]
```diff
@@ -99,7 +99,7 @@ pub enum Error {
     /// The arbitration for the bus was lost.
     ArbitrationLost,
     /// The execution of the I2C command was incomplete.
-    ExecIncomplete,
+    ExecutionIncomplete,
```
Comment (bugadani) [inline esp-hal/src/i2c/master/mod.rs:102]: This needs to be changelogged

File: esp-hal/src/i2c/master/mod.rs [hunk 2]
```diff
@@ -99,7 +99,7 @@ pub enum Error {
...
     /// The number of commands issued exceeded the limit.
     CommandNrExceeded,
```
Comment (bugadani) [inline esp-hal/src/i2c/master/mod.rs:104]: What about Nr?

File: documentation/API-GUIDELINES.md [hunk 3]
```diff
@@ -80,6 +80,8 @@ In general, the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines
 - If a driver only implements a subset of a peripheral's capabilities, it should be placed in the `peripheral::subcategory` module.
   - For example, if a driver implements the slave-mode I2C driver, it should be placed into `i2c::slave`.
   - This helps us reducing the need of introducing breaking changes if we implement additional functionalities.
+- Avoid abbreviations and contractions in the API, where possible.
```
Comment (bugadani) [inline documentation/API-GUIDELINES.md:83]: Cana we proactively say that commonly used ones, like Dma is acceptable?

File: esp-hal/src/i2c/master/mod.rs (hunk 1 above)
Comment (MabezDev) [reply esp-hal/src/i2c/master/mod.rs:102]: I'm on it :)

File: documentation/API-GUIDELINES.md (hunk 3 above)
Comment (MabezDev) [reply documentation/API-GUIDELINES.md:83]: Yes, I'll add a note about that.

File: esp-hal/src/i2c/master/mod.rs [hunk 4]
```diff
@@ -91,17 +91,17 @@ const MAX_ITERATIONS: u32 = 1_000_000;
 #[non_exhaustive]
 pub enum Error {
     /// The transmission exceeded the FIFO size.
-    ExceedingFifo,
+    FifoExceeded,
```
Comment (MabezDev) [inline esp-hal/src/i2c/master/mod.rs:94]: This one is a rust api violation, https://rust-lang.github.io/api-guidelines/naming.html#names-use-a-consistent-word-order-c-word-order.

**Review by bjoernQ — APPROVED** [review]
`Timeout` and `InvalidZeroLength` are not object/verb order but it might sound and look awkward when forced here

I thought there is a list of common abbreviations for use in documentation (like a documentation style guide) we could link to - but I cannot find it

(The approval assumes fixing the errors)

**Comment by jessebraham** [comment]
> `Timeout` and `InvalidZeroLength` are not object/verb order but it might sound and look awkward when forced here

`TimeOut` -> `Timeout` resolves the first one IMO 😁 `ZeroLengthInvalid` doesn't sound too awkward to me, either.

File: esp-hal/MIGRATING-0.22.md [hunk 5]
```diff
@@ -300,3 +300,20 @@ is not compatible with the hardware.
...
+- Error::ExecIncomplete
++ Error::ExecutionIncomplete
+- Error::CommandNrExceeded
++ Error::CommandNumberExceeded
+- Error::ExceedingFifo
++ Error::FifoExceeded
+- Error::TimeOut
++ Error::Timeout
+- Error::InvalidZeroLength
++ Error::ZeroLength
```
Comment (jessebraham) [inline esp-hal/MIGRATING-0.22.md:318]:
```suggestion
+ Error::ZeroLengthInvalid
```

Sorry 😅

Comment (MabezDev) [reply esp-hal/MIGRATING-0.22.md]: Oops! Thanks!

**Review by playfulFence — APPROVED** [review]

File: esp-hal/src/i2c/master/mod.rs (hunk 4 above)
Comment (MabezDev) [reply esp-hal/src/i2c/master/mod.rs:94]: Yes, but only for the drivers we want to stabilize, rmt can stay as is (for now)

**Pull request merged**